src/lib/page_scripts/trade_offers.ts

```diff
@@ -1,15 +1,96 @@
 import {init} from './utils';
 import '../components/trade_offers/better_tracking';
+import '../components/trade_offers/trade_offer_holder_metadata';
 import {inPageContext} from '../utils/snips';
 import {ClientSend} from '../bridge/client';
 import {PingSetupExtension} from '../bridge/handlers/ping_setup_extension';
 import {PingExtensionStatus} from '../bridge/handlers/ping_extension_status';
+import {FetchSteamTrades, FetchSteamTradesResponse} from '../bridge/handlers/fetch_steam_trades';
 
 init('src/lib/page_scripts/trade_offers.js', main);
 
 function main() {}
 
+/**
+ * Gets the trade offers from the local storage or fetches them from the API.
+ * Local storage serves as a cache here.
+ * @param isSentPage if the current page is the sent trade offers page
+ * @returns the trade offers
+ */
+async function fetchTradeOffers(isSentPage: boolean) {
+    const g_steamTrades = JSON.parse(localStorage.getItem('g_steamTrades') || '{}') as FetchSteamTradesResponse;
+    let refetchRequired = true;
+    if (g_steamTrades.sent || g_steamTrades.received) {
+        const latestTradeId = Number.parseInt(g_steamTrades[isSentPage ? 'sent' : 'received']?.[0].offer_id);
+        const latestTradeElement = Number.parseInt(document.querySelector('.tradeoffer')?.id.split('_')[1] ?? '0');
+
+        refetchRequired = Number.isNaN(latestTradeId) || latestTradeId !== latestTradeElement;
+    }
+
+    if (!refetchRequired) {
+        return g_steamTrades;
+    }
+
+    const steamTrades = await ClientSend(FetchSteamTrades, {});
+
+    localStorage.setItem('g_steamTrades', JSON.stringify(steamTrades));
+    return steamTrades;
+}
+
+/**
+ * Fetches the api data for trade offers and stores relevant data in the DOM to be used by Lit components.
+ */
+async function getAndStoreTradeOffers() {
+    const isSentPage = location.pathname.includes('sent');
```

Comment:
I'm trying to think of a good way to avoid having to detect specific pages like this (since URL string matching is somewhat brittle).

For getting the trade offers below, you can just check both `received` and `sent` fields.

However, it seems like we need it to perform the "latest trade offer" check to potentially break the cache. I guess the alternative would be checking if there are any trade offer IDs on the page that don't exist in the API response?

Reply:
We can easily remove the URL check in favor of a more costly check of both the `received` and `sent` fields.

I believe checking only the latest trade offer ID should be enough as older ones are guaranteed to be contained in the API response as a consequence. Although I haven't tested how this behavior changes for large amounts of open trades, in which case we would need extra handling for the refetch anyways though.